Desktop/modules/dynamicmodule.cpp

```diff
@@ -770,26 +641,29 @@ void DynamicModule::setStatus(moduleStatus newStatus)
 	if(_status == moduleStatus::installNeeded && newStatus == moduleStatus::installModPkgNeeded)
 		return;
 
-	_status = newStatus;
+	bool readyForUseInvolved = newStatus == moduleStatus::readyForUse || _status == moduleStatus::readyForUse;
 
-	emit statusChanged();
+	_status = newStatus;
 
 	switch(_status)
 	{
-	case moduleStatus::loadingNeeded:			emit registerForLoading(_name);				break;
-	case moduleStatus::installNeeded:			emit registerForInstalling(_name);			break;
-	case moduleStatus::installModPkgNeeded:		emit registerForInstallingModPkg(_name);	break;
-	case moduleStatus::error:
-		Log::log() << "Just set an error on the status of module "<< _name << std::endl;	break;
+	case moduleStatus::installNeeded:			emit registerForInstalling(_name);			setInstalled(false);					break;
+	case moduleStatus::installModPkgNeeded:		emit registerForInstallingModPkg(_name);	setInstalled(false);					break;
+	case moduleStatus::error:					Log::log() << "Just set an error on the status of module "<< _name << std::endl;	break;
 	default:																				break;
 	}
+
+	emit statusChanged();
+
+	if(readyForUseInvolved)
+		emit readyForUseChanged();
 }
 
 void DynamicModule::reloadDescription()
 {
 	try
 	{
-		loadDescriptionFromFolder(fq(_moduleFolder.absoluteFilePath() + "/" + nameQ() + "/"));
+		loadDescriptionFromFolder(fq(_moduleFolder.absoluteFilePath() + "/" + nameQ() + "/"), false);
 	}
 	catch(std::runtime_error e) { return; } //If it doesnt work then never mind.
```

Comment:
Because it might have been triggered while the user was changing it, saved it with a bunch of crap.
Or maybe it just got removed or whatever. You still don't want the entire application to crash and burn then.